bftclient/include/bftclient/bft_client.h

```diff
@@ -58,8 +60,8 @@ class Client {
   std::string signMessage(std::vector<uint8_t>&);
   void setTransactionSigner(concord::util::crypto::ISigner* signer) { transaction_signer_.reset(signer); }
   // thread safe version of send api
-  Reply sendThreadSafe(const WriteConfig& config, Msg&& request);
-  Reply sendThreadSafe(const ReadConfig& config, Msg&& request);
+  Reply sendThreadSafe(const WriteConfig& config, Msg&& request, bft::communication::NodeNum src_endpoint_num);
+  Reply sendThreadSafe(const ReadConfig& config, Msg&& request, bft::communication::NodeNum src_endpoint_num);
 
  private:
```

Comment:
There is no need to pass src_endpoint_num to the client functions. Each client knows its own principle-id from the configuration it receives in the ctor. This value should be just passed to the communication layer through send() function.

Reply:
done.